Source/objects.cpp

```diff
@@ -3089,7 +3089,7 @@ bool OperateShrineSpiritual(int pnum)
 	for (int8_t &gridItem : player.InvGrid) {
 		if (gridItem == 0) {
 			int r = 5 * leveltype + GenerateRnd(10 * leveltype);
-			uint8_t t = player._pNumInv;
+			Item &invItem = player.InvList[player._pNumInv];
 			player.InvList[t] = GoldTemplate;

```

Comment:
u forgot to change these lines ;p

Reply:
They where not included in the suggestio, but I will do it manually once I'm at a computer.